ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -85,17 +92,27 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
    */
   public List<ResponseInfo> sendAndPoll(List<RequestInfo> requestInfos)
       throws IOException {
-    if (closed) {
-      throw new IllegalStateException("The NetworkClient is closed.");
-    }
-    List<ResponseInfo> responseInfoList = new ArrayList<ResponseInfo>();
-    for (RequestInfo requestInfo : requestInfos) {
-      pendingRequests.add(new RequestMetadata(time.milliseconds(), requestInfo));
+    long startTime = time.milliseconds();
+    try {
+      if (closed) {
+        throw new IllegalStateException("The NetworkClient is closed.");
+      }
+      List<ResponseInfo> responseInfoList = new ArrayList<ResponseInfo>();
+      numPendingConnections.set(pendingRequests.size());
```

Comment:
setting `numPendingConnections` here is unnecessary - it wouldn't have changed at this point. Also why is this called `numPendingConnections`? Is this not `numPendingRequests`?